spec/controllers/saml_idp_controller_spec.rb

```diff
@@ -211,6 +211,29 @@
       end
     end
 
+    context 'service provider is valid' do
+      before do
```

Comment:
I usually see `before` blocks used for **shared** setup. Since this is being used for a single test (same with `before` block below), thoughts in putting it within `it` block instead?